Wrong error message: If QuarkusTest has wrong working directory, it should not blame it on "NoSuchElementException: SRCFG00011: Could not expand value platform.quarkus.native.builder-image in property quarkus.native.builder-image"

To reproduce with quarkus 1.13.3.Final (I haven't tried 2.0 yet):

1. git clone https://github.com/kiegroup/optaplanner-quickstarts
2. Open that project in IntelliJ
3. In the module quarkus-school-timetabling, open the class TimeTableConstraintProviderTest and run it by clicking on the run icon in the gutter of the java file editor at the class declaration line.

Notice that the working directory of the run configuration that IntelliJ gives you by default will be set to "optaplanner-quickstarts" and not "quarkus-school-timetabling". When it's set incorrectly like that, the unit test fails with this error:

```
java.lang.RuntimeException: java.lang.RuntimeException: java.util.NoSuchElementException: SRCFG00011: Could not expand value platform.quarkus.native.builder-image in property quarkus.native.builder-image

    at io.quarkus.test.junit.QuarkusTestExtension.throwBootFailureException(QuarkusTestExtension.java:674)
...
Caused by: java.lang.RuntimeException: java.util.NoSuchElementException: SRCFG00011: Could not expand value platform.quarkus.native.builder-image in property quarkus.native.builder-image
...
 Caused by: java.util.NoSuchElementException: SRCFG00011: Could not expand value platform.quarkus.native.builder-image in property quarkus.native.builder-image
```

To a user, that error message should instead say "your working directory is wrong", so we can fix it and move on.

Here's how to fix it in the run configuration of intellij (which sets the working dir to quarkus-school-timetabling):

![Selection_723](https://user-images.githubusercontent.com/176880/123272369-f9f1f780-d501-11eb-9683-81a0c42893fc.png)

--- a/test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java
+++ b/test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java
@@ -314,6 +314,11 @@ public class QuarkusTestExtension
                         .bootstrap();
             }
 
+            if (curatedApplication.getAppModel().getUserDependencies().isEmpty()) {
+                throw new RuntimeException(
+                        "The tests were run against a directory that does not contain a Quarkus project. Please ensure that the test is configured to use the proper working directory.");
+            }
+
             Index testClassesIndex = TestClassIndexer.indexTestClasses(requiredTestClass);
             // we need to write the Index to make it reusable from other parts of the testing infrastructure that run in different ClassLoaders
             TestClassIndexer.writeIndex(testClassesIndex, requiredTestClass);

